integration/spark/src/test/java/io/openlineage/spark/agent/lifecycle/StaticExecutionContextFactory.java

```diff
@@ -77,16 +80,32 @@ public SparkSQLExecutionContext createSparkSQLExecutionContext(long executionId)
               List<PartialFunction<LogicalPlan, List<OpenLineage.Dataset>>> commonDatasetVisitors =
                   commonDatasetVisitors(sqlContext, sparkContext);
 
-              InputDatasetVisitors inputDatasetVisitors =
-                  new InputDatasetVisitors(commonDatasetVisitors);
-              OutputDatasetVisitors outputDatasetVisitors =
-                  new OutputDatasetVisitors(sqlContext, commonDatasetVisitors);
+              VisitorFactory common =
+                  new CommonVisitorFactory(sqlContext, sparkContext.getJobNamespace());
+              VisitorFactory versionSpecific =
+                  VersionSpecificVisitorsProvider.getInstance(SparkSession.active());
```

Comment:
Rather than having this split here, can we just have the version-specific providers extend the common provider and return a correctly mixed list of input/output visitors?